### PR TITLE
feat: populate cpe_key in organization and product tables

### PR DESCRIPTION
**Pull request by dejanb**

### CODE REVIEW

File: modules/fundamental/src/product/endpoints/test.rs [hunk 1]
```diff
@@ -30,6 +31,7 @@ async fn all_products(ctx: &TrustifyContext) -> Result<(), anyhow::Error> {
             "AMQ Broker",
             ProductInformation {
                 vendor: Some("Red Hat".to_string()),
+                cpe: None,
```
Comment (JimFuller-RedHat) [inline modules/fundamental/src/product/endpoints/test.rs:34]: maybe worthwhile to populate ?

File: modules/ingestor/src/graph/product/mod.rs [hunk 2]
```diff
@@ -120,11 +120,12 @@ impl<'g> ProductContext<'g> {
 #[derive(Clone, Default, Debug)]
 pub struct ProductInformation {
     pub vendor: Option<String>,
+    pub cpe: Option<Cpe>,
 }
 
 impl ProductInformation {
     pub fn has_data(&self) -> bool {
-        self.vendor.is_some()
+        self.vendor.is_some() || self.cpe.is_some()
```
Comment (JimFuller-RedHat) [inline modules/ingestor/src/graph/product/mod.rs:128]: I think idiomatic Rust might look like ` self.vendor.or(self.cpe).is_some()`

File: entity/src/product.rs [hunk 3]
```diff
@@ -9,6 +9,7 @@ pub struct Model {
     pub id: Uuid,
     pub name: String,
     pub vendor_id: Option<Uuid>,
+    pub cpe_key: Option<String>,
```
Comment (JimFuller-RedHat) [inline entity/src/product.rs:12]: have you considered using **cpe** crate (we already have in Cargo.toml)  to also validate cpe with `Cpe::parse(cpe_str)?;` somewhere ?

File: modules/ingestor/src/graph/sbom/spdx.rs [hunk 4]
```diff
@@ -157,7 +159,9 @@ impl SbomContext {
                     "cpe22Type" => match Cpe::from_str(&r.reference_locator) {
                         Ok(cpe) => {
                             refs.push(PackageReference::Cpe(cpe.uuid()));
-                            cpes.add(cpe);
+                            cpes.add(cpe.clone());
+                            // assign only the first cpe for now
```
Comment (JimFuller-RedHat) [inline modules/ingestor/src/graph/sbom/spdx.rs:163]: Can you expand a little bit on rationale ... is there a TODO here ?

**Review by JimFuller-RedHat — APPROVED** [review]
apart from failing CI, made some observations LGTM

File: modules/ingestor/src/graph/product/mod.rs (hunk 2 above)
Comment (dejanb) [reply modules/ingestor/src/graph/product/mod.rs:128]: This pattern is used for all similar code. Let's change them all in a separate PR?

File: modules/fundamental/src/product/endpoints/test.rs (hunk 1 above)
Comment (dejanb) [reply modules/fundamental/src/product/endpoints/test.rs:34]: Actually cpe_key is not propagated to the endpoint at the moment. It used only internally for correlation. Maybe we should wait with introducing it into the api until we finish with todo mentioned below?

Comment (dejanb) [reply modules/fundamental/src/product/endpoints/test.rs:34]: It's being tested in 'service' btw

File: modules/ingestor/src/graph/sbom/spdx.rs (hunk 4 above)
Comment (dejanb) [reply modules/ingestor/src/graph/sbom/spdx.rs]: Add some more context. Thanks

**Review by JimFuller-RedHat — APPROVED** [review]
LGTM

**Pull request merged**